### ISSUE TITLE
Understand Code Logic 

### ISSUE THREAD

**Issue by iOSDevGarg** [Meonardo/janus-videoroom-ios#9]
Can you please help me understand the core Logic of your app?
whenever a new user joins the same room.
are you creating a new Peer connection with local track as yours own track and remote Description of new user?
is it due to multiple users?

**Comment by RockinGarg** [comment]
Would you like to give a try if you have little time. I tried following your approach but fails. I don't have any server I just have firebase integrated. if possible else I am gonna scratch my head on this for sure.

**Comment by Meonardo** [comment]
> Would you like to give a try if you have little time. I tried following your approach but fails. I don't have any server I just have firebase integrated. if possible else I am gonna scratch my head on this for sure.

Can you elaborate your usecase? 
If I understand you correctly, you want to add one or more remote track(s) to the same PeerConnection?


I highly suggest you to build your own janus server, it will help you to understand the way how WebRTC works, 
for example: to check the janus serve log, use the admin api of janus and etc...

**Comment by RockinGarg** [comment]
Mine issue
first one started 
2 connected  - it showed 1st user and self video
1st user - shows self and second user video

3 connected - 
1 - shows all three self and other two video
2 - self and first video (3rd user video no)
3 - self and first user video

I haven't used that publisher subscriber 
All users are on same level and responds to the new user to set offer and answer

**Comment by Meonardo** [comment]
> I haven't used that publisher subscriber

What do you mean? 

> I get a callback from firebase for new user

I did not see any[ transport api ](https://github.com/meetecho/janus-gateway/tree/master/transports)related to `Firebase`.
Any chance you are not using the [janus WebRTC server](https://github.com/meetecho/janus-gateway)?

**Comment by RockinGarg** [comment]
I am using firebase as signaling service instead of sockets to communicate and using default STUN servers for p2p connection.
Is this server code is the entire backend code ? Any API Key is required ?

**Comment by Meonardo** [comment]
Please pay the project website a visit specially the VideoRoom [plugin](https://janus.conf.meetecho.com/videoroomtest.html).

**Comment by fukemy** [comment]
great topic, but you need to know this library work with Janus, so you need to learn how it's work with janus event first to compare with your logic code in Firebase

as i know about this lib, each peerconnection require to handle audio. and video, but only localPeerConnection work with server, if you want to avoid this, just compare the handleID of peerconnection with handleID of roomJanus data

**Comment by fukemy** [comment]
@Meonardo if my explain wrong, please tech me about the problem, thanks

**Comment by Meonardo** [comment]
> @Meonardo if my explain wrong, please tech me about the problem, thanks

You are not wrong, we can find  the [project site](https://janus.conf.meetecho.com/videoroomtest.html) notice
> Notice that this is the original VideoRoom demo, and ***uses a different PeerConnections per each subscription***: if you want to test the new multistream support, instead, try the multistream VideoRoom demo instead. The two demos are interoperable, if you want to see how different subscription mechanisms are used on the same sources.

and the project demo(`janus.js`) file it shows how to [create](https://github.com/meetecho/janus-gateway/blob/943558d761130c35a19cdc3c3a890418f13aa3b8/html/janus.js#L1850) & [save](https://github.com/meetecho/janus-gateway/blob/943558d761130c35a19cdc3c3a890418f13aa3b8/html/janus.js#L1735) `RTCPeerConnection`.

Just for my understanding, 
to understand Janus videoroom plugin simply, forget the WebRTC stack temporarily, 

- to make everyone to see you in the room, publish yourself(create a `RTCPeerConnection` instance and add local a/v track);
- to watch someone in the room, subscribe it(also create a `RTCPeerConnection` instance).
